lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -246,5 +286,79 @@ def cleanup
       @logger.notify "Spent %.2f seconds cleaning up" % (Time.now - start)
     end
 
+    def add_disk(hostname, size)
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/api/v1/vm/' + hostname + '/disk/' + size.to_s)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Post.new(uri.request_uri)
+
+        request['X-AUTH-TOKEN'] = @credentials[:vmpooler_token]
+
+        response = http.request(request)
+      rescue RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        @logger.notify "Failed to connect to vmpooler for additional disk!"
```

Comment:
I think the HTTP rescues are worth keeping to potentially provide info but i added some failures in the provision method.